test/PowerShellEditorServices.Test/Language/SemanticTokenTest.cs

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation.Language;
+using System.Threading.Tasks;
+using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+using Microsoft.PowerShell.EditorServices.Handlers;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document.Proposals;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals;
+using Xunit;
+
+namespace Microsoft.PowerShell.EditorServices.Test.Language
+{
+    public class SemanticTokenTest
+    {
+        [Fact]
+        public async Task MapsFunctionElements()
+        {
+            string text = @"
+function Get-Sum {
+    param( [int]$a, [int]$b )
+    return $a + $b
+}
+";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            foreach(Token t in scriptFile.ScriptTokens)
+            {
+                List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokens.ConvertToSemanticTokens(t));
+                switch(t.Text)
+                {
+                    case "function":
+                    case "param":
+                    case "return":
+                        Assert.Single(mappedTokens);
+                        Assert.Equal(SemanticTokenType.Keyword, mappedTokens[0].Type);
+                        break;
+                    case "Get-Sum":
+                        Assert.Single(mappedTokens);
+                        Assert.Equal(SemanticTokenType.Function, mappedTokens[0].Type);
+                        break;
+                    case "$a":
+                    case "$b":
+                        Assert.Single(mappedTokens);
+                        Assert.Equal(SemanticTokenType.Variable, mappedTokens[0].Type);
```

Comment:
Would it be better to use Assert.Collection here (even though it's only 1 element"

Reply:
Something like 
Assert.Collection(mappedTokens,  sToken => Assert.Equal(SemanticTokenType.Function, sToken.Type));

Reply:
I think if the latter means you can generalise the test code to be more data driven (i.e. add or complexify the test and only need to add more datastructures rather than change code), then it's probably better.

If the array is an implementation detail and you always expect it to only have on element, then the single assertion makes sense